jquery.flot.legend.js

```diff
@@ -395,15 +395,13 @@
         options = plot.getOptions();
 
         plot.hooks.setupGrid.push(function (plot) {
```

Comment:
In insertLegend() is this piece of code:
```js
if (!options.legend.show) {
  return;
}
```

I'm thinking to move those lines here, so basically if the plot is created with legend=false then no hook is being added to ```plot.hooks.setupGrid```.